### ISSUE TITLE
Proxy tests w/ Array filter/map

### ISSUE THREAD

**Issue by jrsearles** [compat-table/compat-table#716]
The ES6 tests for "Array.prototype iteration methods" under "Proxy, internal 'get' calls" are failing for filter & map methods in an environment i am testing. The "constructor" property is also called, which seems correct according to my reading of [ArraySpeciesCreate](http://www.ecma-international.org/ecma-262/6.0/#sec-arrayspeciescreate). 

If you agree, i can submit a PR to update the test.

**Comment by ljharb** [comment]
Which environment?

**Comment by jrsearles** [comment]
A virtual environment i am working on (https://github.com/jrsearles/SandBoxr). I'm trying to suss out whether the test is wrong or my understanding of the spec is wrong.

**Comment by ljharb** [comment]
Yes, I read it the same as you do. All of the methods' [[Get]] traps should be `length`, then `constructor`, then `Symbol.species` (only when the `constructor` isn't `=== Array`), then the rest.
